tests/test_session_0.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# Author: Leonardo Gama (@leogama)
+# Copyright (c) 2008-2016 California Institute of Technology.
+# Copyright (c) 2016-2022 The Uncertainty Quantification Foundation.
+# License: 3-clause BSD.  The full license text is available at:
+#  - https://github.com/uqfoundation/dill/blob/master/LICENSE
+
+from __future__ import print_function
+import dill, sys, __main__
+
+original_modules = set(sys.modules.keys()) - \
+        set(['json', 'urllib', 'xml.sax', 'xml.dom.minidom', 'calendar', 'cmath'])
+original_objects = set(__main__.__dict__.keys())
+original_objects.add('original_objects')
+
+
+# Create various kinds of objects to test different internal logics.
+
+## Modules.
+import json                                         # top-level module
+import urllib as url                                # top-level module under alias
+import test_session_1_byref_false as local_mod      # non-builtin top-level module
```

Comment:
need to clean up any files left on disk (i.e. *pyc or in the cache).

Reply:
These bytecode cache files are really annoying. What do you think of avoiding generating them by calling python in `dill.tests.__main__` with a `-B` flag or a `PYTHONDONTWRITEBYTECODE` env var? It only would be necessary to remove cache files at the end for `__init__.py` and `__main__.py` (which are still hanging there today, no?).

Reply:
Answering my second question, after doing the proper clean up in my test files, the cache files from `__init__.py` and `__main__.py` magically disappeared.